src/lib/table/index.ts

```diff
@@ -451,4 +454,58 @@ export default class Table {
       get: () => this.getRows(filters),
     };
   }
+
+  // Make sure to leave existing columns in case they hold data
+  // If an id is provided, this method will rename the existing column
+  public async updateColumns(
+    schema: Record<
+      string,
+      { id?: string; options?: Partial<MultiSelectOption>[]; type: string }
+    >,
+  ) {
+    const collectionBlocks = await this.queryCollection(this.keys);
+
+    const rawSchema = get(
+      collectionBlocks,
+      `recordMap.collection.${this.keys.collectionId}.value.schema`,
+      {},
+    ) as Schema;
+
+    const newSchema = Object.entries(schema).reduce<Schema>(
+      (newSchema, [key, headingData]) => {
+        let id: string;
+        if (headingData.type === 'title') {
+          id = 'title';
+        } else if (headingData.id) {
+          id = headingData.id;
+        } else {
+          const existing = Object.entries(rawSchema).find(
+            ([id, cell]) => cell.name === key,
+          );
+          id = existing
+            ? existing[0]
+            : crypto.randomFillSync(Buffer.alloc(2)).toString('hex');
+        }
+        newSchema[id] = { name: key, type: headingData.type };
+        if (headingData.options) {
```

Comment:
i'm updating the options for a column here too, starting w/the original list - following my principle of not removing any data by default.